client/view/showStats.js

```diff
@@ -0,0 +1,141 @@
+
+export function updateGlobalCoverageBar(data) {
+  const element = document.getElementById('global-coverage-bar');
+  element.innerHTML = '';
+  data.forEach((item) => {
+    const alpha = 1 - 1/(item.coverage);
+    const itemElem = document.createElement('li');
+    itemElem.classList.add('BrowsersStat__globalCoverageBarItem');
+    itemElem.setAttribute('style', `
+    --p: ${item.coverage};
+    --a: ${alpha};
+    `)
+    if(item.coverage > 10) {
+      itemElem.innerHTML = item.name;
+      itemElem.classList.add('BrowsersStat__globalCoverageBarItem--texted')
+    }
+    element.appendChild(itemElem);
+  })
+}
+
+const statsPlaceholder = document.querySelector('.BrowsersStat__placeholder');
+
+export function hideStatsPlaceholder() {
+  if(statsPlaceholder.classList.contains('BrowsersStat__placeholder--hidden')) {
+    return true;
+  }
+
+  const statsElem = document.querySelector('.BrowsersStat__stat');
+  statsElem.classList.remove('BrowsersStat__stat--hidden');
+
+  statsPlaceholder.classList.add('BrowsersStat__placeholder--hidden');
+
+  return true;
+}
+
+function _getWikipediaLink(id) {
+  return WIKIPEDIA_URL + wikipediaLinks[id]
+}
+
+function _createCoverageCell(coverage) {
+  const coveragePercentageHtmlString = (cov) => cov + '%';
+  const coveragePercentageCssString = (cov) => {
+    const result =  Math.log(1 + cov) * 100 / Math.log(1 + 100);
+    if (result === 0) {
+      return '0';
+    } else if (result > 5) {
+      return result + '%';
+    }
+    return '1px';
+  }
+
+  const coverageCell = document.createElement('td');
+  coverageCell.classList.add('BrowsersStat__td');
+  coverageCell.innerHTML = coveragePercentageHtmlString(coverage);
+  coverageCell.classList.add('BrowsersStat__td--coverage');
+  coverageCell.setAttribute('style', `--c:${coveragePercentageCssString(coverage)}`);
+  return coverageCell;
+}
+
+function _createVersionCell(version) {
+  const versionCell = document.createElement('td');
+  versionCell.classList.add('BrowsersStat__td');
+  versionCell.innerHTML = version;
+  return versionCell;
+}
+
+export function updateBrowsersStats(data) {
+  const element = document.getElementById('browsers-stats');
+
+  const table = document.createElement('table');
+  table.classList.add('BrowsersStat__table');
+
+  data.forEach(({id, name, versions: versionsInput}) => {
+
+    const versions = Object.entries(versionsInput)
+      .sort(([, coverageA], [, coverageB]) => coverageB - coverageA)
+      .map(([ version, coverage ]) => {
+        return {
+          version,
+          coverage,
+        }
+      });
+
+    const tBody = document.createElement('tbody');
+    const tr = document.createElement('tr');
+    tr.classList.add('.BrowsersStat_tr');
+
+    const iconCell = document.createElement('td');
+    iconCell.classList.add('BrowsersStat__td');
+    const iconElem = document.createElement('img');
+    iconElem.classList.add('BrowsersStat__icon');
+    iconElem.src = `/assets/browser_logos/${id}.svg`;
```

Comment:
We need to process these files with Vite to have cache busters. There is a way to “import” file as a string with a pth.

```js
import imgUrl from './img.svg'
```

Reply:
How can i import a file, which name i know only in run time?...

Reply:
Import all icons to a variables and then use.

It also will force us to think about the case when Can I Use will add a new browser and we will not have icon for it.

Reply:
Let's make it TODO. Ok?

Reply:
Sure https://github.com/browserslist/browserl.ist/issues/317